vochain/scrutinizer/vote.go

```diff
@@ -59,7 +59,18 @@ func (s *Scrutinizer) GetEnvelope(nullifier []byte) (*models.VoteEnvelope, []byt
 	if envelope == nil {
 		return nil, nil, fmt.Errorf("transaction is not an Envelope")
 	}
-	return envelope, stx.Signature, nil
+	envelope.Nullifier = voteRef.Nullifier
+	return &types.EnvelopePackage{
+		Nonce:                envelope.Nonce,
+		ProcessId:            envelope.ProcessId,
+		VotePackage:          envelope.VotePackage,
+		Nullifier:            nullifier,
+		EncryptionKeyIndexes: envelope.EncryptionKeyIndexes,
+		Weight:               voteRef.Weight.Bytes(),
+		TxIndex:              voteRef.TxIndex,
```

Comment:
Do we need index and height? Isn't TxHash enough?

Reply:
We have `getBlockByHash` and `getEnvelope` by nullifier, but not `getTxByHash`. This is partially because we don't have an index of transactions, so we need to fetch a transaction directly from the block & index.

If we don't store `txIndex` and `height`, there's no way to tell the user which transaction contained their vote, which imo is necessary.